src/app/content/utils/archiveTreeUtils.ts

```diff
@@ -11,8 +11,13 @@ import {
 } from '../types';
 import { getIdVersion, stripIdVersion } from './idUtils';
 
+// This is exported so it can be cleared for tests cases
+export const CACHED_FLATTENED_TREES = new Map<string, Array<LinkedArchiveTree | LinkedArchiveTreeSection>>();
 export function flattenArchiveTree(tree: LinkedArchiveTree): Array<LinkedArchiveTree | LinkedArchiveTreeSection> {
-  return [tree, ...flatten(tree.contents.map((section) =>
+  if (CACHED_FLATTENED_TREES.has(tree.id)) {
+    return CACHED_FLATTENED_TREES.get(tree.id)!;
```

Comment:
I think the preferred way of doing it is `assertDefined` but it shouldn't make a difference

Reply:
I'm always feeling weird when doing `assertDefined()` in cases when I've just checked for it, but there is one advantage: it is harder to overlook when changing the code